querydsl-lucene3/src/main/java/com/querydsl/lucene3/LuceneSerializer.java

```diff
@@ -120,6 +120,12 @@ private Query toQuery(Operation<?> operation, QueryMetadata metadata) {
             @SuppressWarnings("unchecked") //This is the expected type
             Query rv = ((Constant<Query>) operation.getArg(0)).getConstant();
             return rv;
+        } else if (op == LuceneOps.BOOST) {
+            BoostElement element = (BoostElement) operation;
```

Comment:
this is not a safe cast, since the mixin type is just an operation, you need to get the predicate and boost factor from the operation